### ISSUE TITLE
Safari CORS problem

### ISSUE THREAD

**Issue by thienpow** [ipfs/js-ipfs#3154]
if i am running the webpage on http://localhost:3000 it works for both Safari & Chrome & Firefox...
however if i am running the webpage on https://mydomain.com only Safari fail to connect to the go-ipfs daemon while all the other browser still work on the https://mydomain.com.  I think the CORS header at the api got a different requirements or format that Safari ask for.

and actually it happen the same for https://webui.ipfs.io/ too... safari 13.1.1 would be rejected to gain access to the local go-ipfs daemon

<img width="1487" alt="Screenshot 2020-07-08 at 10 23 31 PM" src="https://user-images.githubusercontent.com/9796623/86930539-baec5a00-c169-11ea-8fff-26fe2254e818.png">

in the mean time, Brave browser just works
<img width="1172" alt="Screenshot 2020-07-08 at 10 24 51 PM" src="https://user-images.githubusercontent.com/9796623/86930760-eff8ac80-c169-11ea-8003-26949b7ed46b.png">

in short, http://localhost:xxxx works on all browser, https://xxx.com only Safari would fail

and i think it's a very old problem for long...

https://stackoverflow.com/questions/16824661/cors-request-not-working-in-safari

_Originally posted by @thienpow in https://github.com/ipfs/js-ipfs/issues/3130#issuecomment-655547248_

**Comment by thienpow** [comment]
https://www.chromestatus.com/feature/4539473312350208

i think this could be the problem, bcos i find that the api in the browser would generate a "same-origin" header... this obviously won't allow a webapp deployed on https to connect a local machine http://localhost:5001

**Comment by achingbrain** [comment]
Does this happen against js-ipfs too?

**Comment by thienpow** [comment]
<img width="1680" alt="Screenshot 2020-07-09 at 7 56 32 AM" src="https://user-images.githubusercontent.com/9796623/86981717-f9aa0080-c1b9-11ea-9ee2-16404b22f8ce.png">

just tried shooting up jsipfs daemon... same problem for Safari + Https...
localhost:5002/webui is working on all browser...
https://webui.ipfs.io only Safari would fail, while all other browser is working.

**Comment by jacobheun** [comment]
@lidel is this related to the localhost secure issues?

**Comment by lidel** [comment]
I believe this is unrelated: seems Safari's CORS  validation follows different rules, 
but I don't own a Mac to confirm if the issue with CORS on https://webui.ipfs.io is real.

@rafaelramalho19 – mind taking a look?

**Comment by rafaelramalho19** [comment]
@lidel I can confirm that I can replicate this issue on Safari only 
![image](https://user-images.githubusercontent.com/24696635/87705836-6d3da600-c796-11ea-92af-40ae9ccb6319.png)

This appears to be an old issue with Safari, and I couldn't find a direct approach to fixing this (https://stackoverflow.com/questions/39076180/https-page-was-not-allowed-to-run-insecure-content-from-safari-extension-uri)

**Comment by achingbrain** [comment]
This is not to do with CORs, this is because we're trying to access content in an insecure way from a secure location - the page is loaded over https but the API request is made over http.

You used to be able to turn this off as a Safari preference but it's been removed, which is probably for the best.

I'm going to close this, it's a duplicate of https://github.com/ipfs-shipyard/ipfs-webui/issues/1541

**Comment by lidel** [comment]
Clarification for drive-by reader: all other browsers (Chromium, Firefox) mark `127.0.0.1` as [Secure Context][]. 
This is [part of W3C's Secure Context spec](https://www.w3.org/TR/secure-contexts/#localhost) which Safari is not following on purpose.

[Secure Context]: https://developer.mozilla.org/en-US/docs/Web/Security/Secure_Contexts
